src/main/java/com/minecolonies/coremod/commands/CommandEntryPointNew.java

```diff
@@ -0,0 +1,596 @@
+package com.minecolonies.coremod.commands;
+
+import com.minecolonies.api.configuration.Configurations;
+import com.minecolonies.coremod.colony.permissions.ForgePermissionNodes;
+
+import net.minecraft.command.CommandBase;
+import net.minecraft.command.CommandException;
+import net.minecraft.command.ICommandSender;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.server.MinecraftServer;
+import net.minecraft.util.math.BlockPos;
+import net.minecraftforge.server.permission.PermissionAPI;
+import net.minecraftforge.server.permission.context.PlayerContext;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.jetbrains.annotations.NotNull;
+
+import javax.annotation.Nullable;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * Command entry point to make minecraft inheritance happy.
+ */
+public class CommandEntryPointNew extends CommandBase
+{
+    @NotNull
+    private final TreeNode<Menu> root;
+
+    private static class ParsingResult
+    {
+        @NotNull private final List<String> tabCompletions;
+        @NotNull private final TreeNode<Menu> executionTreeNode;
+        @Nullable private final List<ActionArgument> executionActionArgumentList;
+        @Nullable private String badArgument;
+
+        ParsingResult(@NotNull final List<String> tabCompletions, @NotNull final TreeNode<Menu> executionTreeNode,
+                @Nullable final List<ActionArgument> executionActionArgumentList,
+                @Nullable final String badArgument)
+        {
+            super();
+            this.tabCompletions = tabCompletions;
+            this.executionTreeNode = executionTreeNode;
+            this.executionActionArgumentList = executionActionArgumentList;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Store a copy of "executionActionArgumentList". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2384)